### DISCUSSION TITLE
Can useMatches return empty array?

### DISCUSSION THREAD

**Discussion by jansedlon** [remix-run/remix#10001] [Q&A]
Hi, from time to time, I receive a notification my sentry that a call in remix-i18next crashed.

It crashes in the following code

```typescript
function useLocale() {
  const matches = useMatches();
  const rootMatch = matches.at(0);
}
````

The line where `matches.at(0)`.

Is there a scenario in which useMatches can return empty array?

**Comment by kiliman** [comment]
Are you sure you have a root loader?

Can you create a simple repro in StackBlitz or CodeSandbox?

**Reply by jansedlon** [reply]
My codebase is 120k LOC, it has a root loader 😅 It usually crashes in 2 or more nested layout deep route. Each with a loader. No one else has reported this problem in remix-i18next so it must be some kind of edge case, not sure i would be able to make a repro. I can try though

**Reply by jansedlon** [reply]
The latest error occured [here](https://moje.flixy.cz/bud_uspesna/p/vytvo-sv-j-e-book-raz-dva--hr59kbgu8). It's not in english but note that this page has at least 2 loader. One for the root, one for the current route. And this occurrence particularly crashed in a component that contains the Stripe form. So I don't think there's a chance that loader is missing

<img width="829" alt="image" src="https://github.com/user-attachments/assets/c5d0f5d5-f2fb-4aa0-a6e7-b6dcfcdaf03a">

**Reply by jansedlon** [reply]
So actually the error `.at() is not a function` doesn't mean that there's a code working on undefined data but that `.at` is not supported in many browser like Instagram and Facebook browsers that are before certain version. I failed to understand the error 😅
